### DISCUSSION TITLE
Server User / Policy to allow only one specifc user ID

### DISCUSSION THREAD

**Discussion by JanMalch** [supabase/supabase#11084] [Questions] [answered]
Hi, I have the following use-case: **An unauthenticated user performs a request to my own API. Based on that the server does some computations and has to insert data.**

What's the best way to solve this without the `service_role` API key (to be safe from accidental leaks)?

---

I thought of having a user just for my server. So the server logs in with credentials from the `.env` file and then performs the insert.
But I wasn't able to write a policy that only allows this one user to perform the insert. I tried

```
CREATE POLICY "Enable insert for server user only" ON "public"."my_table"
AS PERMISSIVE FOR INSERT
TO authenticated
WITH CHECK (auth.uid() = "<my-server-user-uid>")
```

but that fails because `"<my-server-user-uid>"` is not a column.

Thanks in advance!

**Comment by GaryAustin1** [comment, marked as answer]
In Postgres you use single quotes for a string... `'my_uuid'`.
Note you probably will need `'my_uuid'::uuid` to force the type.
Double quotes are used for "Non Standard names" of columns, tables, functions, etc.
